app/navigation/DocumentsNavigator.scala

```diff
@@ -85,26 +89,34 @@ class DocumentsNavigator @Inject()(implicit frontendAppConfig: FrontendAppConfig
     case _ => routes.SessionExpiredController.onPageLoad()
   }
 
-
   private def whenGoverningDocumentApprovedPageNav(userAnswers: UserAnswers, mode: Mode): Call = userAnswers.get(WhenGoverningDocumentApprovedPage) match {
     case Some(_) if mode == CheckMode => regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
     case Some(_) => regulatorDocsRoutes.IsApprovedGoverningDocumentController.onPageLoad(mode)
     case _ => routes.SessionExpiredController.onPageLoad()
   }
 
   private def isApprovedGoverningDocumentPageNav(userAnswers: UserAnswers, mode: Mode): Call = userAnswers.get(IsApprovedGoverningDocumentPage) match {
+    case Some(true) if mode == CheckMode => userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage) match {
+      case Some(_) if userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage).isDefined =>
+        regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
+      case _ => regulatorDocsRoutes.HasCharityChangedPartsOfGoverningDocumentController.onPageLoad(mode)
+    }
     case Some(true) => regulatorDocsRoutes.HasCharityChangedPartsOfGoverningDocumentController.onPageLoad(mode)
     case Some(false) => regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
     case _ => routes.SessionExpiredController.onPageLoad()
   }
 
-  private def hasCharityChangedPartsOfGoverningDocumentPageNav(userAnswers: UserAnswers, mode: Mode): Call = userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage) match {
-    case Some(true) => regulatorDocsRoutes.SectionsChangedGoverningDocumentController.onPageLoad(mode)
+  private def hasCharityChangedPartsOfGoverningDocumentPageNav(userAnswers: UserAnswers, mode: Mode): Call =
+    userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage) match {
+    case Some(true) if mode == CheckMode && userAnswers.get(SectionsChangedGoverningDocumentPage).isDefined =>
+      regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
+    case Some(true)  => regulatorDocsRoutes.SectionsChangedGoverningDocumentController.onPageLoad(mode)
     case Some(false) => regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
```

Comment:
might be a little cleaner if `Some(true) if ...` and `Some(false)` are collapsed into case x | y? not too bothered about this one though

Reply:
if is only applicable for some (true) hence not combined